### ISSUE TITLE
../src/test/test-alloc-util.c:69:9: runtime error: load of address 0x60200000003c with insufficient space for an object of type 'int'

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#13026]
**systemd version the issue has been seen with**
> latest master

**Used distribution**
 > Fedora 29 with gcc 9.1.1

Yet another one:

```
$ build/test-alloc-util
../src/test/test-alloc-util.c:69:9: runtime error: load of address 0x60200000003c with insufficient space for an object of type 'int'
0x60200000003c: note: pointer points here
  03 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
              ^ 
    #0 0x4024cb in test_memdup_multiply_and_greedy_realloc ../src/test/test-alloc-util.c:69
    #1 0x4024cb in main ../src/test/test-alloc-util.c:155
    #2 0x7f72c69d1412 in __libc_start_main (/lib64/libc.so.6+0x24412)
    #3 0x40129d in _start (/home/mrc0mmand/repos/systemd/build/test-alloc-util+0x40129d)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/test/test-alloc-util.c:69:9 in
```

I wanted to wait until I find more of them, but wasn't lucky so far.

**Comment by evverx** [comment]
> I wanted to wait until I find more of them, but wasn't lucky so far.

They are all over the place: https://github.com/systemd/systemd/pull/12771#issuecomment-502139157, https://github.com/systemd/systemd/pull/12812#issuecomment-502780455 and https://github.com/systemd/systemd/pull/12826#issuecomment-503259185 so I'm not sure we need more of them :-)

**Comment by poettering** [comment]
Should be fixed by #13034
